lib/ecto/query.ex

```diff
@@ -529,6 +529,34 @@ defmodule Ecto.Query do
 
       from query, group_by: ^[:some_field, dynamic(...)]
 
+  ## `select` and `select_merge`
+
+  Dynamics can be interpolated inside values in maps at the root of a
+  `select` or `select_merge`. For example, you can write:
+
+      select = %{
+        period: dynamic([p], p.month),
+        metric: dynamic([p], p.distance)
+      }
+
+      from query, select: ^select
+
+  As with `where` and friends, it is not possible to pass dynamics
+  outside of a root. For example, this won't work:
+
+      from query, select: %{field: ^dynamic(...)}
+
+  But this will:
+
+      from query, select: ^%{field: dynamic(...)}
+
+  Maps with dynamics can also be merged into existing `select` structures,
+  enabling a variety of possibilities for partially dynamic selects:
+
+      metric = dynamic([p] p.distance)
+
+      from query, select: {:map, [:id, :period]}, select_merge: ^metric
```

Comment:
```suggestion
      from query, select: [:id, :period], select_merge: ^metric
```

?

Reply:
Updated in [cb6a223](https://github.com/elixir-ecto/ecto/pull/3927/commits/cb6a223d0e6734d5d192c21465dd1acea5e83a8d).